src/AutoRest.CSharp/Mgmt/Output/ResourceOperation.cs

```diff
@@ -91,7 +96,12 @@ internal class ResourceOperation : TypeProvider
 
         public IEnumerable<ResourceListMethod>? ManagementGroupExtensionListMethods => _managementGroupExtensionsListMethods ??= GetManagementGroupExtensionsListMethods();
 
-        public virtual ClientMethod? GetMethod => _getMethod ??= Methods.FirstOrDefault(m => m.Name.StartsWith("Get") && m.RestClientMethod.Responses[0].ResponseBody?.Type.Name == ResourceData.Type.Name);
+        public virtual ClientMethod? GetMethod => _getMethod ??= Methods.FirstOrDefault(m => IsGetResourceMethod(m) && m.RestClientMethod.Parameters.FirstOrDefault()?.Name.Equals("scope") == true) ?? Methods.OrderBy(m => m.Name.Length).FirstOrDefault(m => IsGetResourceMethod(m));
```

Comment:
Can we find a way to share the code from here with https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Mgmt/Output/MgmtRestClient.cs#L27?